### PR TITLE
SNOW-1491325 AST for `session.generator` and `session.range`

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1491325

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Generate AST for `session.generator` and `session.range`.

### CODE REVIEW

File: tests/ast/data/session_range.test [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+r = session.range(1, 10, 2)
+
+r = session.range(1, None)
+
+r = session.range(1, None, 2)
+
+## EXPECTED OUTPUT
+
+res1 = session.range(start=10, stop=None, step=1)
```
Comment (sfc-gh-azwiegincew) [inline tests/ast/data/session_range.test:15]: We probably shouldn't generate kwargs in the unparser in this case.

**Review by sfc-gh-azwiegincew — APPROVED** [review]

File: tests/ast/data/session_range.test (hunk 1 above)
Comment (sfc-gh-vbudati) [reply tests/ast/data/session_range.test:15]: I merged the corresponding PR in the other codebase, should I remove the kwargs from there?

Comment (sfc-gh-azwiegincew) [reply tests/ast/data/session_range.test:15]: Yep.

Comment (sfc-gh-vbudati) [reply tests/ast/data/session_range.test]: Here's the PR for that: https://github.com/snowflakedb/snowflake/pull/198094

**Review by sfc-gh-azwiegincew — APPROVED** [review]

**Pull request merged**